labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -23,7 +23,13 @@ object PatternMatchingExercise {
    *************************************************************************/
 
   def describeLanguage(s: String): String = {
-    error("fix me")
+    s match {
+      case "Java" | "Smalltalk" => "OOP"
+      case "Clojure" | "Haskell" => "Functional"
+      case "Scala" => "Hybrid"
+      case "C" => "Procedural"
+      case _ => "Unknown"
+    }
   }
```

Comment:
Sure they are. They’re accomplishing the same thing as the curly braces in the following C code do:

```c
if (x < 7) {
	printf("small");
} else {
	printf("big");
}
```

That is, they help ensure that the programmer and compiler agree about how to parse the code, even if that code changes.